tools/get-deps/Makefile

```diff
@@ -7,7 +7,7 @@ EXECUTABLE=get-deps
 all: $(EXECUTABLE)
 	go mod tidy
 	go mod vendor
-	CGO_ENABLED=0 go build -o $(EXECUTABLE) $(SOURCES)
+	GOOS=$(LOCAL_GOOS) CGO_ENABLED=0 go build -o $(EXECUTABLE) $(SOURCES)
```

Comment:
Where is `LOCAL_GOOS` defined? And why would we need it? If I am building on mac, wouldn't it build for that GOOS by default?

I see `LOCAL_GOOS` defined in the parent Makefile, but would that even carry through? It is not exported.

Reply:
Sorry forgot the root Makefile change, pushed it up.  For GOOS: it's exported in the root makefile and set to linux so we need to override it back to the local one.

Reply:
There is a part of me that wonders if this belongs in [eve-build-tools](https://github.com/lf-edge/eve-build-tools). But on second thought, it belongs here. This is not a tool used shared in different repos, at later build stages. This is basic for Makefile. So it stays here.

Reply:
@andrewd-zededa I finally figured out what is bothering me. Let's say I go into `tools/get-deps` and do `make all`. By default, it will build for the right OS or wrong OS? Based on `GOOS=$(LOCAL_GOOS)`, that implies that it will do so only if `LOCAL_GOOS` is set? I would think the simple default case would be `cd tools/get-deps && make` (or `make -C tools/get-deps`) should build it for whatever OS I am on.

I think you are saying, because the root Makefile overrides `GOOS` at various places, we do not want that to happen when called from there, so we run it as `make -C tools/get-deps LOCAL_GOOS=$(LOCAL_GOOS)`.

If that is the case, should it be simpler? Leave `tools/get-deps/Makefile` alone, and change the root `Makefile` to be:

```make
$(GET_DEPS):
	$(MAKE) -C $(GET_DEPS_DIR) GOOS=$(LOCAL_GOOS)
```

That way:

* if I just run `make -C tools/get-deps` it builds for whatever I am on (normal expectations)
* if I run it from within root `Makefile`, because it is the Makefile itself that changes `GOOS`, it is the same Makefile that sets it back for calling the tool

Would that work?

Reply:
Yeah I see your concern, I'll make the change